### PR TITLE
chore: remove react-scripts and jest

### PR DESCRIPTION
**Pull request by ZachJW34**
### User facing changelog
na

### Additional details
Remove jest from the monorepo and redundant webpack-preprocessor tests.

We already have system-tests for webpack-preprocessor that extend webpack. We should avoid having packages that install test dependencies (like react-scripts which was including jest) and utilize system-tests.

Size of node_modules before/after change:

```
Before: 2610 MB
After: 2446 MB
```

### Steps to test
na

### How has the user experience changed?
na

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: .circleci/workflows.yml [hunk 1]
```diff
@@ -1740,31 +1740,6 @@ jobs:
...
-      - run:
-          name: Types ts-loader
-          command: yarn types
-          working_directory: npm/webpack-preprocessor/examples/use-ts-loader
-      - run:
-          name: Test ts-loader
-          command: yarn test
-          working_directory: npm/webpack-preprocessor/examples/use-ts-loader
-      - run:
-          name: Start React app
```
Comment (lmiller1990) [inline .circleci/workflows.yml:1794]: Looks good, I looked at system tests, I see 

image


We've also got a few system-tests using `.babelrc` so I don't think we are losing any coverage.

**Review by lmiller1990 — APPROVED** [review]

File: npm/eslint-plugin-dev/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "lint": "eslint --ext .js,json,.eslintrc .",
     "lint-changed": "node ./lib/scripts/lint-changed",
     "lint-fix": "npm run lint -- --fix",
-    "test": "jest"
```
Comment (emilyrohrbough) [inline npm/eslint-plugin-dev/package.json:10]: 🎉

File: npm/webpack-preprocessor/.eslintignore [hunk 3]
```diff
@@ -5,5 +5,4 @@
 **/cypress/fixtures
 **/test/fixtures
 **/_test-output
-/cypress/tests/e2e/compile-error.js
-/examples/use-babelrc/cypress/e2e/spec.cy.js
```
Comment (emilyrohrbough) [inline npm/webpack-preprocessor/.eslintignore:9]: We might want to keep these .gitingnore paths around? Stale branches pulling in this change will likely might have these git ignore files laying around and git will suggest we check them in?

File: npm/webpack-preprocessor/.gitignore [hunk 4]
```diff
@@ -4,5 +4,4 @@ node_modules/
 _test-output
 **/cypress/videos
 cypress/screenshots
-examples/react-app/build/
```
Comment (emilyrohrbough) [inline npm/webpack-preprocessor/.gitignore]: We might want to keep these .gitingnore paths around? Stale branches pulling in this change will likely might have these git ignore files laying around and git will suggest we check them in?

File: npm/webpack-preprocessor/README.md [hunk 5]
```diff
@@ -42,12 +42,6 @@ module.exports = (on) => {
 }
 ```
 
-## Examples
```
Comment (emilyrohrbough) [inline npm/webpack-preprocessor/README.md:45]: Would it be worth listing the system-test examples we have in place?

File: npm/webpack-preprocessor/package.json [hunk 6]
```diff
@@ -59,9 +59,6 @@
     "mocha": "^7.1.0",
     "mockery": "2.1.0",
     "proxyquire": "2.1.3",
```
Comment (emilyrohrbough) [inline npm/webpack-preprocessor/package.json:53]: Can we also drop `@babel/plugin-proposal-nullish-coalescing-operator`? It was only used in `use-babel` example AFAIK

Just looking through, maybe out of scope, but looks like these might not be used anymore in this module:
- `@fellow/eslint-plugin-coffee`
- `@typescript-eslint/eslint-plugin`
- `@typescript-eslint/parse`

Might be a few others as well that can now rely on the root level dev deps.

**Review by emilyrohrbough — COMMENTED** [review]
Vite-dev-server's .estlintrc file includes
```
"globals": {
    "jest": "readonly"
  },
```
this might also be stale given there are no jest tests in vite-dev-server.

**Review by emilyrohrbough — APPROVED** [review]

File: npm/webpack-preprocessor/package.json (hunk 6 above)
Comment (ZachJW34) [reply npm/webpack-preprocessor/package.json:53]: I'll take a look later today, I love removing unused deps!

Comment (emilyrohrbough) [reply npm/webpack-preprocessor/package.json:53]: me too 😄

File: npm/webpack-preprocessor/.gitignore (hunk 4 above)
Comment (ZachJW34) [reply npm/webpack-preprocessor/.gitignore]: I'll revert! Not harming anyone and less confusion for devs.

File: npm/webpack-preprocessor/README.md (hunk 5 above)
Comment (ZachJW34) [reply npm/webpack-preprocessor/README.md:45]: I've added a few examples back.

File: npm/webpack-preprocessor/package.json (hunk 6 above)
Comment (ZachJW34) [reply npm/webpack-preprocessor/package.json:53]: Cleaned up a few more! Could spend a week+ cleaning up all of the unused deps but I just focused on webpack-preprocessor (and batteries included).

Comment (lmiller1990) [reply npm/webpack-preprocessor/package.json:53]: There is a bunch of packages to automate removed old deps eg https://www.npmjs.com/package/depcheck. Probably worth doing at some point, should be easy enough to run a few and see what it recommends removing.

**Pull request merged**